autopilot/contractor.go

```diff
@@ -258,8 +258,11 @@ func (c *contractor) performContractMaintenance(ctx context.Context, w Worker) (
 	c.cachedMinScore = minScore
 	c.mu.Unlock()
 
+	// create a new ip filter
+	ipFilter := newIPFilter(c.logger)
```

Comment:
Might be good to mention here that this is done to keep already resolved hostnames cached for the duration of the maintenance iteration.